spec/support/fake_ansible_repo.rb

```diff
@@ -0,0 +1,357 @@
+      #
+      def parse_file_struct(file_struct)
+        yaml_data = file_struct.gsub(/^(\s*)([a-zA-Z0-9\.])/, '\1- \2')
+                               .gsub(/\/$/, '/:')
+        YAML.load(yaml_data)
+      end
+
+      # Builds a (nested) directory structure at the given path and file
+      # structure definition.
+      #
+      # This method is called recursively, so `repo_relpath` and `file_struct`
+      # might be in a nested partial state depending how deep the call of this
+      # method has made it.
+      #
+      def build_files_at(repo_rel_path, file_struct)
+        file_struct.each do |entry|
+          case entry
+          when Array  then build_files_at(repo_rel_path, entry)
+          when Hash   then build_directory(repo_rel_path, entry)
+          when String then build_file(repo_rel_path, entry)
+          end
+        end
+      end
+
+      # Generates a single directory (hash key)
+      #
+      # If there is entries (value), it defers back to `.build_files_at`,
+      # otherwise if it is `nil` it represents a empty dir.
+      #
+      def build_directory(repo_rel_path, file_struct)
+        file_struct.each do |dir, entries|
+          new_dir = repo_rel_path.join(dir.sub(/\/$/, ''))
+
+          FileUtils.mkdir_p(new_dir)
+          build_files_at(new_dir, entries) if entries
+        end
+      end
+
+      VAR_FILE_FNMATCHES = [
+        "**/extra_vars.{yml,yaml}",
+        "**/roles/*/vars/*.{yml,yaml}",
+        "**/roles/*/defaults/*.{yml,yaml}"
+      ]
+
+      # Gerates a single file
+      #
+      # If it is a reserved file name/path, then the contents are generated,
+      # otherwise it will be a empty file.
+      #
+      def build_file(repo_rel_path, entry)
+        full_path = repo_rel_path.join(entry)
+
+        File.open(full_path, "w") do |new_file|
+          case
+          when filepath_match?(full_path, "**/requirements.{yml,yaml}")
+            new_file.puts REQUIREMENTS
+          when filepath_match?(full_path, *VAR_FILE_FNMATCHES)
+            new_file.puts VAR_DATA
+          when filepath_match?(full_path, "**/roles/*/meta/main.{yml,yaml}")
+            new_file.puts META_DATA
+          when filepath_match?(full_path, "**/*.{yml,yaml}")
+            new_file.puts dummy_playbook_data_for(full_path.basename)
+          end
+        end
+      end
+
+      # Given a collection of glob based `File.fnmatch` strings, confirm
+      # whether any of them match the given path.
+      def filepath_match? path, *acceptable_matches
+        acceptable_matches.any? { |match| path.fnmatch?(match, File::FNM_EXTGLOB) }
+      end
+
+      # Init new repo at local_repo
+      #
+      #   $ cd /tmp/clone_dir/hello_world_local && git init .
+      #
+      def git_init
+        @repo  = Rugged::Repository.init_at(repo_path.to_s)
+        @index = repo.index
+      end
+
+      # Add new files to index
+      #
+      #   $ git add .
+      #
```

Comment:
Also seems unnecessary, though I see you are doing this for completeness, since other methods need it, so I'm good